### PR TITLE
Add single read

### PR DESCRIPTION
**Pull request by anujdeshpande**

### CODE REVIEW

File: arducam_mega.c [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+
+
+	LOG_INF("Initializing the camera");
+	spi = device_get_binding(SPI_DEV_NAME);
+	if (!device_is_ready(spi)) {
+		LOG_ERR("Device SPI not ready, aborting test");
+		return -1;
+	}
+
+	/* struct arducam_mega_config *cfg = dev->config; */
```
Comment (Bucknalla) [inline arducam_mega.c]: Can we remove comments like this before we go to merge the initial release?

File: arducam_mega.c [hunk 2]
```diff
@@ -0,0 +1,466 @@
+#include "arducam_mega.h"
```
Comment (Bucknalla) [inline arducam_mega.c:23]: Have you run this through `clang-format` yet?

**Review by Bucknalla — CHANGES_REQUESTED** [review]
Looks good generally, we'll merge this one and leave the burst open while we wait to solve the weird issues we're having.

**Comment by Bucknalla** [comment]
Also please do a rebase and squash :)

**Comment by anujdeshpande** [comment]
@Bucknalla Can you take a look again? Have
- squashed
- run clang-format
- removed commented code

**Comment by Bucknalla** [comment]
@anujdeshpande can you please push the `clang-format` config file you're using? We should match the zephyr one

File: arducam_mega.c [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+      fs_write(&file, imageBuff, i);
+
+      fs_close(&file);
+      i = 0;
+      break;
+    }
+  }
+}
+
+int arducam_mega_take_picture(CAM_IMAGE_MODE mode,
```
Comment (Bucknalla) [inline arducam_mega.c]: What is the difference between 'take' and 'save' photo?

File: arducam_mega.c [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+  gpio_pin_toggle_dt(&spec);
+  received_length -= 1;
+  return rxdata[0];
+}
+
+void camera_save_fifo(const char *base_path, uint32_t length, char *filename) {
+
+  uint8_t imageData = 0;
+  uint8_t imageDataNext = 0;
+  uint8_t headFlag = 0;
```
Comment (Bucknalla) [inline arducam_mega.c]: Styling inconsistency -> uint8_t imageData, imageDataNext, headFlag = 0

File: arducam_mega.c [hunk 5]
```diff
@@ -0,0 +1,367 @@
+#include "arducam_mega.h"
+#include <zephyr/device.h>
```
Comment (Bucknalla) [inline arducam_mega.c:24]: Missing license

File: arducam_mega.h [hunk 6]
```diff
@@ -0,0 +1,323 @@
+#ifndef __ARDUCAM_MEGA_H__
```
Comment (Bucknalla) [inline arducam_mega.h:23]: Missing license

**Review by Bucknalla — CHANGES_REQUESTED** [review]

File: arducam_mega.c (hunk 3 above)
Comment (anujdeshpande) [reply arducam_mega.c]: Take keeps it on the cameras buffer

When the app starts I take a photo to warm up the camera - but I don't save it. After that I take and save photos as per the config

Comment (Bucknalla) [reply arducam_mega.c]: Perhaps we can tidy this up so we share some code between the two functions, so save utilises take but with extra logic to grab the photos after capture

Comment (Bucknalla) [reply arducam_mega.c]: I also this 'take' -> 'capture' it's a bit more clear and explains the whole 'save' / 'capture' difference

File: arducam_mega.c (hunk 4 above)
Comment (anujdeshpande) [reply arducam_mega.c]:
```
/home/anuj/arribada/arribada-workspace/time-lapse-zephyr.git/arducam-driver-zephyr/arducam_mega.c:154:28: note: 'imageDataNext' was declared here
  154 |         uint8_t imageData, imageDataNext, headFlag = 0;
```

File: arducam_mega.c [hunk 7]
```diff
@@ -0,0 +1,398 @@
...
+
+uint8_t camera_get_bit(uint8_t addr, uint8_t bit)
+{
+	uint8_t temp;
+	temp = camera_read_reg(addr);
+	temp = temp & bit;
+	return temp;
+}
+
+uint8_t cameraReadByte()
```
Comment (Bucknalla) [inline arducam_mega.c]: camera_read_byte()

File: arducam_mega.c [hunk 8]
```diff
@@ -0,0 +1,398 @@
...
+			fs_write(&file, imageBuff, i);
+
+			fs_close(&file);
+			i = 0;
+			break;
+		}
+	}
+}
+
+int arducam_mega_capture_picture(CAM_IMAGE_MODE mode, CAM_IMAGE_PIX_FMT pixel_format)
```
Comment (Bucknalla) [inline arducam_mega.c]: arducam_mega_capture_image

Let's use image instead of picture

File: arducam_mega.c [hunk 9]
```diff
@@ -0,0 +1,398 @@
...
+	return length;
+}
+int arducam_mega_save_picture(char *filename, const char *mount_point, int image_length)
+{
+	LOG_INF("Saving picture\n");
+	camera_save_fifo(mount_point, image_length, filename);
+	return 0;
+}
+
+int arducam_mega_get_cameraid()
```
Comment (Bucknalla) [inline arducam_mega.c]: arducam_mega_get_id

**Review by Bucknalla — CHANGES_REQUESTED** [review]
Almost there! Just a couple of tiny language points that make things a little more uniform.

File: arducam_mega.c (hunk 8 above)
Comment (Bucknalla) [reply arducam_mega.c]: Across other functions too

**Comment by Bucknalla** [comment]
Merging this so we have a working main, will ideally merge bursting at some point in the future

**Pull request merged**